src/main/java/org/opensearch/security/identity/SecurityIndices.java

```diff
@@ -0,0 +1,166 @@
+                    if (xcp.currentName() != META) {
+                        xcp.nextToken();
+                        xcp.skipChildren();
+                    } else {
+                        while (xcp.nextToken() != XContentParser.Token.END_OBJECT) {
+                            if (xcp.currentName().equals(SCHEMA_VERSION)) {
+
+                                Integer version = xcp.intValue();
+                                if (version < 0) {
+                                    version = NO_SCHEMA_VERSION;
+                                }
+                                return version;
+                            } else {
+                                xcp.nextToken();
+                            }
+                        }
+
+                    }
+                }
+                xcp.nextToken();
+            }
+            return NO_SCHEMA_VERSION;
+        } catch (Exception e) {
+            // since this method is called in the constructor that is called by OpenSearchSecurityPlugin.createComponents,
+            // we cannot throw checked exception
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected boolean doesScheduledJobIdentityIndexExists() {
+        if (!cs.state().metadata().hasConcreteIndex(SCHEDULED_JOB_IDENTITY_INDEX)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Create scheduled job identity index.
+     *
+     * @param actionListener action called after create index
+     */
+    public void initScheduledJobIdentityIndex(ActionListener<CreateIndexResponse> actionListener) {
+        try {
+            CreateIndexRequest request = new CreateIndexRequest(SCHEDULED_JOB_IDENTITY_INDEX)
+                    .mapping(getScheduledJobIdentityMappings(), XContentType.JSON);
+            request.settings(
+                Settings
+                    .builder()
+                    // Schedule job identity index is small. 1 primary shard is enough
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    // Job scheduler puts both primary and replica shards in the
+                    // hash ring. Auto-expand the number of replicas based on the
+                    // number of data nodes (up to 20) in the cluster so that each node can
+                    // become a coordinating node. This is useful when customers
+                    // scale out their cluster so that we can do adaptive scaling
+                    // accordingly.
+                    // At least 1 replica for fail-over.
+                    .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, minJobIndexReplicas + "-" + maxJobIndexReplicas)
+                    .put("index.hidden", true)
+            );
+            client.admin().indices().create(request, markMappingUpToDate(SecurityIndex.SCHEDULED_JOB_IDENTITY, actionListener));
```

Comment:
is there a chance that this will fail with `already exists` error?

Reply:
Before this function is called the code checks the cluster state to see if the index already exists: https://github.com/opensearch-project/security/pull/2773/files#diff-c56e4aaf50cd20ad928dbb7dc5026e5b2ab1fcb92396290ff4a880203dc220d1R52-R54

Reply:
aah.. got it,